### PR TITLE
extract the scheduler from handler package

### PR DESCRIPTION
**Pull request by keis**
Introduces a new Scheduler interface that holds ScheduleTask

### CODE REVIEW

**Comment by alde** [comment]
i like it

**Comment by keis** [comment]
First attempt just crashed because of how the scheduler was initialised.

**Comment by keis** [comment]
Did a quick test and now things work. Will you throw a goats eye @alde ?

File: routes/routes.go [hunk 1]
```diff
@@ -48,18 +62,3 @@ func notFound(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNotFound)
 	json.NewEncoder(w).Encode(nil)
 }
-
-var routes = types.Routes{
```
Comment (alde) [inline routes/routes.go]: this means we have to explicitly add the test if we add another route

But that might actually be a good thing

**Comment by alde** [comment]
approve

LGTM

**Pull request merged**